VERSIONING.md

```diff
@@ -0,0 +1,220 @@
+        * Experimental modules will start their versioning at `v0.0.0` and will
+          increment their minor version when backwards incompatible changes are
+          released and increment their patch version when backwards compatible
+          changes are released.
+        * All stable modules that use the same major version number will use the
+          same entire version number.
+            * Stable modules may be released with an incremented minor or patch
+              version even though that module has not been changed, but rather so
+              that it will remain at the same version as other stable modules that
+              did undergo change.
+            * When an experimental module becomes stable a new stable module version
+              will be released and will include this now stable module. The new
+              stable module version will be an increment of the minor version number
+              and will be applied to all existing stable modules as well as the newly
+              stable module being released.
+* Versioning of the associated [contrib
+  repository](https://github.com/open-telemetry/opentelemetry-collector-contrib) of
+  this project will be idiomatic of a Go project using [Go
+  modules](https://github.com/golang/go/wiki/Modules).
+    * [Semantic import
+      versioning](https://github.com/golang/go/wiki/Modules#semantic-import-versioning)
+      will be used.
+        * Versions will comply with [semver 2.0](https://semver.org/spec/v2.0.0.html).
+        * If a module is version `v2` or higher, the
+          major version of the module must be included as a `/vN` at the end of the
+          module paths used in `go.mod` files (e.g., `module
+          github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sprocessor/v2`, `require
+          github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sprocessor/v2 v2.0.1`) and in the
+          package import path (e.g., `import
+          "github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sprocessor/v2"`). This includes
+          the paths used in `go get` commands (e.g., `go get
+          github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sprocessor/v2@v2.0.1`.  Note there
+          is both a `/v2` and a `@v2.0.1` in that example. One way to think about
+          it is that the module name now includes the `/v2`, so include `/v2`
+          whenever you are using the module name).
+        * If a module is version `v0` or `v1`, do not include the major version
+          in either the module path or the import path.
+    * In addition to public APIs, telemetry produced by stable instrumentation
+      will remain stable and backwards compatible. This is to avoid breaking
+      alerts and dashboard.
+    * Modules will be used to encapsulate receivers, processor, exporters,
+      extensions, and any other independent sets of related components.
```

Comment:
We currently use one component - one module approach. Does this deviate from it so that multiple components can be on one module?

Reply:
I think that the package structure currently used in the contrib repo would make it difficult to group related sets of different types of components in a single module, but it would certainly be possible to group multiple related processors or extensions in a single module.  That said, I think that package structure is independent of versioning policy and the use of modules as the unit of API versioning.  Here we're stating that unrelated components should live in separate modules.

Reply:
I think that can be changed if you ask me, I would like to see a better proposal for contrib structure, but we can discuss that later.